datasets/xsum_factuality/xsum_factuality.py

```diff
@@ -0,0 +1,182 @@
+Neural abstractive summarization models are highly prone to hallucinate content that is unfaithful to the input
+document. The popular metric such as ROUGE fails to show the severity of the problem. The dataset consists of
+faithfulness and factuality annotations of abstractive summaries for the XSum dataset. We have crowdsourced 3 judgements
+ for each of 500 x 5 document-system pairs. This will be a valuable resource to the abstractive summarization community.
+"""
+
+_HOMEPAGE = "https://research.google/tools/datasets/xsum-hallucination-annotations/"
+
+_LICENSE = "https://creativecommons.org/licenses/by/4.0/"
+
+_URL = "https://raw.githubusercontent.com/google-research-datasets/xsum_hallucination_annotations/master/"
+_URLs = {
+    "factuality": _URL + "factuality_annotations_xsum_summaries.csv",
+    "hallucination": _URL + "hallucination_annotations_xsum_summaries.csv",
+}
+
+
+class XsumFactualityConfig(datasets.BuilderConfig):
+    """BuilderConfig for XsumFactuality"""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for XsumFactuality.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(XsumFactualityConfig, self).__init__(**kwargs)
+
+
+class XsumFactuality(datasets.GeneratorBasedBuilder):
+    """XSum Hallucination Annotations: Faithfulness and factuality annotations of XSum summaries"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        XsumFactualityConfig(
+            name="xsum_factuality",
+            version=datasets.Version("1.1.0"),
+            description="Raters are shown the news article and the system summary, and are tasked with "
+            "identifying and annotating the spans that aren't supported by the input article.",
+        ),
+        XsumFactualityConfig(
+            name="xsum_faithfulness",
+            version=datasets.Version("1.1.0"),
+            description="Raters are shown the news article and the hallucinated system summary, and are "
+            "tasked with assessing the summary whether it is factual or not.",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "xsum_factuality"
+
+    def _info(self):
+        if self.config.name == "xsum_factuality":
+            features = datasets.Features(
+                {
+                    "bbcid": datasets.Value("int32"),
+                    "system": datasets.Value("string"),
+                    "summary": datasets.Value("string"),
+                    "is_factual": datasets.ClassLabel(names=["yes", "no", "NULL"]),
+                    "worker_id": datasets.Value("string"),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "bbcid": datasets.Value("int32"),
+                    "system": datasets.Value("string"),
+                    "summary": datasets.Value("string"),
+                    "hallucination_type": datasets.ClassLabel(names=["intrinsic", "extrinsic", "NULL"]),
```

Comment:
```suggestion
                    "hallucination_type": datasets.ClassLabel(names=["intrinsic", "extrinsic"]),
```
same here

Reply:
Thanks. I will make the necessary changes soon.

Reply:
@lhoestq I have made all the suggested changes.